src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts

```diff
@@ -392,9 +404,10 @@ export class TableComponent implements AfterContentChecked, OnInit, OnChanges, O
     if (!this.data) {
       return; // Wait for data
     }
-    this.rows = [...this.data];
     if (this.search.length > 0) {
       this.updateFilter();
+    } else {
+      this.rows = [...this.data];
```

Comment:
While I was on it I already did the other one too ;) As mentioned before only a suggestion how to write the test.

```diff
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.spec.ts b/src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.spec.ts
index 19c59e5e22..8d9f385a3c 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.spec.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.spec.ts
@@ -94,6 +94,16 @@ describe('TableComponent', () => {
       component.updateFilter(true);
     };
 
+    it('should not replace rows twice', () => {
+      fixture.detectChanges();
+      component.useData();
+      const begin = [...component.rows];
+      component.search = 'sth';
+      spyOn(component, 'updateFilter').and.stub();
+      component.useData();
+      expect(begin).toEqual(component.rows);
+    });
+
     it('should find a particular number', () => {
       expectSearch('5', [{ a: 5, b: 50, c: true }]);
       expectSearch('9', [{ a: 9, b: 90, c: true }]);
```